postgres/datadog_checks/postgres/metadata.py

```diff
@@ -106,12 +233,221 @@ def report_postgres_metadata(self):
         }
         self._check.database_monitoring_metadata(json.dumps(event, default=default_json_event_encoding))
 
+        elapsed_s_schemas = time.time() - self._time_since_last_schemas_query
+        if elapsed_s_schemas >= self.schemas_collection_interval and self._collect_schemas_enabled:
+            metadata = self._collect_schema_info()
+            event = {
+                "host": self._check.resolved_hostname,
+                "agent_version": datadog_agent.get_version(),
+                "dbms": "postgres",
+                "kind": "pg_databases",
+                "collection_interval": self.schemas_collection_interval,
+                "dbms_version": self._payload_pg_version(),
+                "tags": self._tags_no_db,
+                "timestamp": time.time() * 1000,
+                "metadata": metadata,
+                "cloud_metadata": self._config.cloud_metadata,
+            }
+            json_event = json.dumps(event, default=default_json_event_encoding)
+            self._log.debug("Reporting the following payload for schema collection: {}".format(json_event))
+            self._check.database_monitoring_metadata(json_event)
+
     def _payload_pg_version(self):
         version = self._check.version
         if not version:
             return ""
         return 'v{major}.{minor}.{patch}'.format(major=version.major, minor=version.minor, patch=version.patch)
 
+    def _collect_schema_info(self):
+        databases = []
+        if self._check.autodiscovery:
+            databases = self._check.autodiscovery.get_items()
+        elif self._config.dbname != 'postgres':
+            databases.append(self._config.dbname)
+        else:
+            # if we are only connecting to 'postgres' database, not worth reporting data model
+            return
+
+        metadata = []
+        self._log.debug("Collecting schema metadata...")
```

Comment:
That debug statement isn't terribly informative as it is but I could see it having some value if we include the list of databases the metadata collection is going to go over.

Reply:
oh totally right, I forgot to remove this while I was actively debugging myself, meant to remove